### PR TITLE
Allows to view top 20 active users

### PR DESCRIPTION
**Pull request by hectorcorrea**

### CODE REVIEW

**Comment by awead** [comment]
@hectorcorrea tests? You could throw in a view test, which are fast, or test the response from the solr query.

File: app/controllers/admin/stats_controller.rb [hunk 1]
```diff
@@ -14,8 +14,9 @@ def index
     end
 
     # Query Solr for top 5 depositors
+    depositor_count = top_depositors_count()
```
Comment (mjgiarlo) [inline app/controllers/admin/stats_controller.rb]: I think you can dump this variable and just call the method directly.

File: app/controllers/admin/stats_controller.rb [hunk 2]
```diff
@@ -14,8 +14,9 @@ def index
...
     depositor_key = Solrizer.solr_name('depositor', :stored_searchable, type: :string)
-    top_depositors_url = "#{ActiveFedora.solr_config[:url]}/terms?terms.fl=#{depositor_key}&terms.sort=count&terms.limit=5&wt=json&omitHeader=true"
+    top_depositors_url = "#{ActiveFedora.solr_config[:url]}/terms?terms.fl=#{depositor_key}&terms.sort=count&terms.limit=#{depositor_count}&wt=json&omitHeader=true"
```
Comment (mjgiarlo) [inline app/controllers/admin/stats_controller.rb]: Just call top_depositors_count here.

File: app/controllers/admin/stats_controller.rb [hunk 3]
```diff
@@ -40,4 +41,13 @@ def index
 
     render 'index'
   end
+
+  private 
+  def top_depositors_count
+    count = params[:dep_count].to_i 
+    if count < 5 || count > 20
```
Comment (mjgiarlo) [inline app/controllers/admin/stats_controller.rb]: In Rails, you can use the handy-dandy `#in?` method with a range here. So you could replace this method with:

``` ruby
def top_depositors_count
  count = params[:dep_count].to_i
  count.in?(5..20) ? count : 5
end
```

**Comment by hectorcorrea** [comment]
@awead @mjgiarlo Added a view test and implemented the code tweaks (no hash rockets were introduced in the production of this PR)

File: spec/views/admin/stats/index.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    before do
+      assign(:files_count, {})
+      assign(:top_formats, [])
+      assign(:users_stats, {})
+      assign(:recent_users, [])
+      assign(:active_users, top_20_active_users)
+      params[:dep_count] = 20
+      render
+    end
+    it "should show top 25 depositors, without an option to view more" do
```
Comment (mjgiarlo) [inline spec/views/admin/stats/index.html.erb_spec.rb]: `s/25/20/` ?

Comment (hectorcorrea) [inline spec/views/admin/stats/index.html.erb_spec.rb]: aaaaaaargggggg..., yes 20. fix coming up.

**Comment by mjgiarlo** [comment]
> @awead @mjgiarlo Added a view test and implemented the code tweaks (no hash rockets were introduced in the production of this PR)

:laughing: :bomb: :hash: :rocket: :boom: 

I made one comment on what may be a typo but it's small :fries: so :+1: :clap:

**Comment by hectorcorrea** [comment]
Typo has been fixed.

**Comment by mjgiarlo** [comment]
Woot.  Thoughts, merges, @cam156 @awead ?

**Comment by awead** [comment]
:+1:

**Pull request merged**